StRoot/StETofCalibMaker/StETofCalibMaker.cxx

```diff
@@ -2608,3 +2679,301 @@ StETofCalibMaker::writeHistograms()
+	 
+	 //steering vecs
+	 if(i == get4IdVec.at(key).at(n)){ 
+	   mStateVec[i].push_back(stateVec.at(key).at(n));
+	   mStartVec[i].push_back(startVec.at(j));	   
+	 } 	 
+       }  
+     }
+   }
+   
+   //set map validity check evtids ... EvtZero states only valid to first change of state on any get4
+   mStateMapStart = 0 ;
+   mStateMapStop  = startVec.at(0);
+   mDbEntryStart   = startVec.at(0);
+   mDbEntryStop    = startVec.at((startVec.size()-1));
+
+
+   if(fileZero){
+     mDbEntryStart   = 0;
+   }
+  
+   sort( mMasterStartVec.begin(), mMasterStartVec.end() );
+   mMasterStartVec.erase( unique( mMasterStartVec.begin(), mMasterStartVec.end() ), mMasterStartVec.end() );
+
+ }
+
+// -------------------------------------------------------------------------------
+
+void StETofCalibMaker::checkGet4State(unsigned long int eventNr){
+
+  if(eventNr >= mStateMapStart && eventNr < mStateMapStop) {
+    return; // stateMap still valid 
+  }
+
+  unsigned long int closestStop  = 99999999;
+  unsigned long int closestStart = 0;
+
+  //loop over stateMap
+
+  for(unsigned int i =0; i< 1728; i++){
+
+    std::vector<unsigned long int> tmpStart = mStartVec[i];
+    std::vector<short>             tmpState = mStateVec[i];
+
+    //find closest evtNr & state for each Get4
+    unsigned int      indexStart = 0;
+    // int               indexStop  = 0;
+    short             newState   = 0;
+    // unsigned long int newStart   = 0;
+    // unsigned long int newStop    = 0;
+
+    if(tmpStart.size() == 0 ) continue;
+    
+    auto lower = std::lower_bound(tmpStart.begin(), tmpStart.end(), eventNr);
+    indexStart = std::distance(tmpStart.begin(), lower);    
+    if(indexStart > 0) indexStart--;
+
+    //event past last change on get4 in entry
+    if(eventNr > tmpStart.at((tmpStart.size() -1))){
+      indexStart = (tmpStart.size() -1);
+    }
+  
+    if((indexStart < (tmpStart.size() -1 )) && eventNr == tmpStart.at(indexStart + 1)){
+	indexStart++;	
+    }
+    
+    //get new state and push to map
+    newState = tmpState.at(indexStart);
+
+    if(tmpStart.at(indexStart) > eventNr ) newState = mGet4ZeroStateMap[i];
+
+    mGet4StateMap[i] = newState;
+     
+  } //Get4 Loop
+
+  //  bool Found=false;
+    for(unsigned int z=0; z< mMasterStartVec.size();z++){
+
+      if(z==0){
+	closestStart = 0;
+	closestStop  = mMasterStartVec.at(z);
+	continue;
+      }
+
+      if(z != (mMasterStartVec.size()-1)){
+	
+	if(eventNr == mMasterStartVec.at(z) ){
+	  closestStart = mMasterStartVec.at(z);
+	  closestStop  = mMasterStartVec.at(z+1);
+	  //	  Found = true;
+	  break;
+	}else if(eventNr < mMasterStartVec.at(z+1) && eventNr > mMasterStartVec.at(z)){	
+	  closestStart = mMasterStartVec.at(z);
+	  closestStop  = mMasterStartVec.at(z+1);
+	  //  Found = true;
+	  break;
+	}
+
+      }else{ // last interval
+	closestStart = mMasterStartVec.at(z);
+	closestStop  = 99999999;
+	//	Found = true;
+	break;
+      }
```

Comment:
Looks like a little re-arrangement could remove some of those `continue` and `break` lines. No `break` should be needed on the last interval either, I expect.

I'm also doubtful that the assignment on the first interval (z==0) is necessary, since you always continue the loop in that case and always decide on new values for closestStart and closestStop by the last interval, with the exception of the situation where `mMasterStartVec` has a size of `1`. But is that ever the situation? And even then, would the values assigned for the last interval be sufficient?
```suggestion
      if(z == 0){ // first interval
	closestStart = 0;
	closestStop  = mMasterStartVec.at(z);

      } else if(z == (mMasterStartVec.size()-1)){ // last interval
        closestStart = mMasterStartVec.at(z);
        closestStop  = 99999999;
        // Found = true;

      } else if(eventNr == mMasterStartVec.at(z) ||
          (eventNr < mMasterStartVec.at(z+1) && eventNr > mMasterStartVec.at(z))){
        closestStart = mMasterStartVec.at(z);
        closestStop  = mMasterStartVec.at(z+1);
        //  Found = true;
        break;
      }
```
or perhaps just this is enough with no assignment for z==0? You decide.
```suggestion
      if(z == (mMasterStartVec.size()-1)){ // last interval
        closestStart = mMasterStartVec.at(z);
        closestStop  = 99999999;
        // Found = true;

      } else if(z!=0 &&
          (eventNr == mMasterStartVec.at(z) ||
          (eventNr < mMasterStartVec.at(z+1) && eventNr > mMasterStartVec.at(z)))){
        closestStart = mMasterStartVec.at(z);
        closestStop  = mMasterStartVec.at(z+1);
        //  Found = true;
        break;
      }
```

Reply:
The zero case was relevant for files containing only a few events (and could thus end up with a mMasterStartVec of size one) at some point, or when during the state determination step some "early" events where not taken into account due to a job crashing.
Likely not needed anymore but I would like to keep it  just in case.